mlflow/tracking/fluent.py

```diff
@@ -127,21 +127,23 @@ def start_run(run_uuid=None, experiment_id=None, source_name=None, source_versio
         if source_name is not None:
             user_specified_tags[MLFLOW_SOURCE_NAME] = source_name
         if source_type is not None:
-            user_specified_tags[MLFLOW_SOURCE_TYPE] = source_type
+            user_specified_tags[MLFLOW_SOURCE_TYPE] = SourceType.to_string(source_type)
         if source_version is not None:
             user_specified_tags[MLFLOW_GIT_COMMIT] = source_version
         if entry_point_name is not None:
             user_specified_tags[MLFLOW_PROJECT_ENTRY_POINT] = entry_point_name
 
         tags = context.resolve_tags(user_specified_tags)
 
+        # Polling resolved tags for run meta data : source_name, source_version,
+        # entry_point_name, and source_type which is store in RunInfo for backward compatibility.
         active_run_obj = MlflowClient().create_run(
             experiment_id=exp_id_for_run,
             run_name=run_name,
-            source_name=tags.get(MLFLOW_SOURCE_NAME),  # For backwards compatability
```

Comment:
Micronit: I actually kinda like the "For backwards compatibility" comments b/c they help clarify what we might be able to remove with MLflow 1.0, but ok w/ deleting them too

Reply:
You are right! Done ... added smaller annotation to avoid text wrapping.